### PR TITLE
Pr add domu v1

### PR DESCRIPTION
**Pull request by andr2000**

### CODE REVIEW

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-kernel/kernel-module-gles/kernel-module-gles_1.9.bbappend [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require inc/xt_shared_env.inc
+
+PVRKM_URL = "git://git@gitpct.epam.com/epmd-aepr/pvr_km_vgpu_img.git"
```
Comment (lorc) [inline recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-kernel/kernel-module-gles/kernel-module-gles_1.9.bbappend]: I think, this is the same value as in `kernel-module-gles_1.9.bb`

File: recipes-dom0/dom0-image-thin-initramfs/dom0-image-thin-initramfs.bbappend [hunk 2]
```diff
@@ -15,6 +15,8 @@ python __anonymous () {
         d.appendVarFlag("do_compile", "depends", " domu-image-fusion:do_${BB_DEFAULT_TASK} ")
     if "domr" in guests :
         d.appendVarFlag("do_compile", "depends", " domu-image-litmusrt:do_${BB_DEFAULT_TASK} ")
+    if "domu" in guests :
```
Comment (lorc) [inline recipes-dom0/dom0-image-thin-initramfs/dom0-image-thin-initramfs.bbappend]: I think it is time to convert this succession of `if`s to something more data-driver:

```
guest_depends = {'domu':'domu-image-weston', 'domr':'domu-image-litmusrt`, ....}
for guest, dep in gust_depends.items():
   if guest in guests:
       d.appendVarFlag("do_compile", "depends", " " + dep  + ":do_${BB_DEFAULT_TASK} ")
```

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/images/core-image-thin-initramfs.bbappend [hunk 3]
```diff
@@ -25,6 +25,8 @@ python __anonymous () {
         d.appendVar("IMAGE_INSTALL", " domf domf-run domf-install-artifacts")
     if "domr" in guests :
         d.appendVar("IMAGE_INSTALL", " domr domr-run domr-install-artifacts")
+    if "domu" in guests :
```
Comment (lorc) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/images/core-image-thin-initramfs.bbappend]: The same. Actually, we can have `guest_depends` in some global variable, to eliminate  this copy-pasted code.

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7795-salvator-xs-domu.dts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+	passthrough {
+		compatible = "simple-bus";
+		ranges;
+		#address-cells = <2>;
+		#size-cells = <2>;
+
+		gsx: gsx@fd000000 {
+			compatible = "renesas,gsx";
+			reg = <0 0xfd000000 0 0x3ffff>;
+			/* interrupts = <GIC_SPI 119 IRQ_TYPE_LEVEL_HIGH>;*/
```
Comment (lorc) [inline recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7795-salvator-xs-domu.dts]: Do we need commented out strings?

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas_%.bbappend [hunk 5]
```diff
@@ -24,12 +24,14 @@ SRC_URI_append_salvator-xs-h3-xt = " \
     file://r8a7795-salvator-xs-dom0.dts;subdir=git/arch/${ARCH}/boot/dts/renesas \
     file://r8a7795-salvator-xs-domd.dts;subdir=git/arch/${ARCH}/boot/dts/renesas \
     file://r8a7795-salvator-xs-doma.dts;subdir=git/arch/${ARCH}/boot/dts/renesas \
+    file://r8a7795-salvator-xs-domu.dts;subdir=git/arch/${ARCH}/boot/dts/renesas \
```
Comment (lorc) [inline recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas_%.bbappend]: I think, we need to reduce amount of copy-paste somehow.

File: recipes-domu/domu-image-weston/domu-image-weston.bbappend [hunk 6]
```diff
@@ -0,0 +1,70 @@
+FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
+FILESEXTRAPATHS_prepend := "${THISDIR}/../../inc:"
+FILESEXTRAPATHS_prepend := "${THISDIR}/../../recipes-domx:"
+
+do_configure[depends] += "domd-image-weston:do_domd_install_machine_overrides"
+
+SRC_URI = " \
+    repo://github.com/iartemenko/manifests;protocol=https;branch=pr_testv3;manifest=prod_gen3_test/domu.xml;scmdata=keep \
```
Comment (lorc) [inline recipes-domu/domu-image-weston/domu-image-weston.bbappend]: I thought, that should be located in xen-troops project...

File: recipes-domu/domu-image-weston/files/meta-xt-prod-extra/recipes-graphics/images/core-image-weston.bbappend [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    kernel-modules \
+    kmscube \
+"
+
+# Do not support secure environment
+IMAGE_INSTALL_remove = " \
+    optee-linuxdriver \
+    optee-linuxdriver-armtz \
+    optee-client \
+    libx11-locale \
```
Comment (lorc) [inline recipes-domu/domu-image-weston/files/meta-xt-prod-extra/recipes-graphics/images/core-image-weston.bbappend]: libx11-locale has nothing to do with optee

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-kernel/kernel-module-gles/kernel-module-gles_1.9.bbappend (hunk 1 above)
Comment (andr2000) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-kernel/kernel-module-gles/kernel-module-gles_1.9.bbappend]: fixed now the base .bb, so these are different

File: recipes-dom0/dom0-image-thin-initramfs/dom0-image-thin-initramfs.bbappend (hunk 2 above)
Comment (andr2000) [reply recipes-dom0/dom0-image-thin-initramfs/dom0-image-thin-initramfs.bbappend]: this is a refactoring which needs to be done later

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/images/core-image-thin-initramfs.bbappend (hunk 3 above)
Comment (andr2000) [reply recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/images/core-image-thin-initramfs.bbappend]: this is a refactoring which needs to be done later

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas_%.bbappend (hunk 5 above)
Comment (andr2000) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas_%.bbappend]: this is a refactoring which needs to be done later

File: recipes-domu/domu-image-weston/domu-image-weston.bbappend (hunk 6 above)
Comment (andr2000) [reply recipes-domu/domu-image-weston/domu-image-weston.bbappend]: good catch, will update

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7795-salvator-xs-domu.dts (hunk 4 above)
Comment (andr2000) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7795-salvator-xs-domu.dts]: will remove

File: recipes-domu/domu-image-weston/files/meta-xt-prod-extra/recipes-graphics/images/core-image-weston.bbappend (hunk 7 above)
Comment (andr2000) [reply recipes-domu/domu-image-weston/files/meta-xt-prod-extra/recipes-graphics/images/core-image-weston.bbappend]: done

**Comment by lorc** [comment]
`Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

**Pull request merged**